src/Features/ERDDAP/Platform/Grabber/index.tsx

```diff
@@ -2,46 +2,60 @@
  * Component to retrieve platform from store and pass to children as render prop
  */
 import * as React from "react"
-import { connect } from "react-redux"
 
-import { StoreState } from "Shared/constants/store"
-import { UnitSystem } from "Features/Units/types"
-
-import { PlatformFeatureWithDatasets } from "../../types"
+import { usePlatforms } from "../../hooks"
+import { PlatformFeature, PlatformFeatureCollection } from "../../types"
 
 export interface Props {
-  /** selected platfrom id */
+  /** selected platform id */
   platformId: string
-  children(props: RenderProps): JSX.Element
-}
-
-export interface ReduxProps {
-  platforms: PlatformFeatureWithDatasets[]
-  unit_system: UnitSystem
+  children: (props: RenderProps) => JSX.Element
 }
 
 export interface RenderProps {
-  platform: PlatformFeatureWithDatasets
-  unit_system: UnitSystem
+  platform: PlatformFeature
 }
 
-function mapStateToProps({ erddap, unit }: StoreState): ReduxProps {
-  return {
-    platforms: erddap.platforms,
-    unit_system: unit.system
+export const ErddapPlatformGetter: React.FunctionComponent<Props> = ({ platformId, children }) => {
+  const { isLoading, data } = usePlatforms()
+
+  if (isLoading) {
+    return <h4>Loading platform data</h4>
   }
-}
 
-export class ErddapPlatformGetterBase extends React.Component<Props & ReduxProps, object> {
-  public render() {
-    const platforms = this.props.platforms.filter(p => (p.id as string) === this.props.platformId)
+  if (data) {
+    const platform = (data as PlatformFeatureCollection).features.filter((p) => (p.id as string) === platformId)
 
-    if (platforms.length > 0) {
-      return this.props.children({ platform: platforms[0], unit_system: this.props.unit_system })
+    if (platform.length > 0) {
+      return children({ platform: platform[0] })
     } else {
-      return <p>Unable to load platform {this.props.platformId}</p>
+      return <p>Unable to load platform {platformId}</p>
     }
   }
+
+  return <h4>Error</h4>
+}
+
+interface PlatformsRenderProps {
+  platforms: PlatformFeature[]
 }
 
-export const ErddapPlatformGetter = connect(mapStateToProps)(ErddapPlatformGetterBase)
+interface PlatformsProps {
+  children: (props: PlatformsRenderProps) => JSX.Element
```

Comment:
Codacy found an issue: ['JSX' is not defined.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)